### PR TITLE
Reader post spotlight indexing

### LINKED ISSUE #8785: Spotlight Integration: Index downloaded items in reader
Using the `CoreSpotlight` API, index a reader items as they are downloaded.

* Indexed items are private (should be by default with this API)
* Index any likes on a reader post
* We probably want to have an aggressive expiration date in this situation (as compared to other indexed items).
* Use the same identifier for `uniqueIdentifier` on `CSSearchableItem` when indexing the same item as `NSUserActivity` API in order to avoid duplicates and boost rankings.

Ref: #8783

### LINKED ISSUE #8786: Spotlight Integration: Add analytics
Title has it. Let's track when a user enters the app via a spotlight search.

Ref: #8781, #8782, #8783, #8785

### PR DESCRIPTION
**Pull request by bummytime**
reader_indexing

This PR builds on top of the work done in  #8781 by indexing and posts opened in the reader. Unlike post/page indexing we only index posts if the user drills into a specific post (vs. scrolling through a topic tableview). In addition, the reader post `CSSearchableItem` is set to expire after 7 days.

(Note that featured image thumbnails are out of scope for this PR)

Fixes #8785 
Fixes #8786 

**To test:**
_You will want to test this PR on a device because the sim is not reliable when working with spotlight._

Should be straight-forward to test:
1. Open Me -> App Settings and clear the spotlight index
2. Browse around in reader and drill into a few articles. 
3. Search in spotlight for various keywords verifying articles show up
4. Tap on search results and make sure appropriate article appears in the reader

@frosty would you mind taking a peek?

### CODE REVIEW

File: WordPress/Classes/Models/ReaderPost+Searchable.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+extension ReaderPost: SearchableItemConvertable {
+    var searchItemType: SearchItemType {
+        return .readerPost
+    }
+
+    var isSearchable: Bool {
+        guard let title = searchTitle, !title.isEmpty else {
+            // If the title is empty or nil, don't index it
```
Comment (frosty) [inline WordPress/Classes/Models/ReaderPost+Searchable.swift]: Can we not use a snippet of the test as a title?

File: WordPress/Classes/Models/ReaderPost+Searchable.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        return readerPostPreview
+    }
+
+    var searchKeywords: [String]? {
+        return generateKeywordsFromContent()
+    }
+
+    var searchExpirationDate: Date? {
+        let sevenDaysFromNow = Calendar.current.date(byAdding: .day, value: 7, to: Date())
```
Comment (frosty) [inline WordPress/Classes/Models/ReaderPost+Searchable.swift]: Should we add 1 .weekOfYear instead of 7 days? Maybe it doesn't matter.

**Comment by bummytime** [comment]
@frosty thanks for the quick review!!

> I spotted a functional issue on iPad

Good catch. On it!

> I was also wondering if we should index Reader posts publicly where possible?

Since we are using CoreSpotlight in this PR, the indexes [can only be private](https://developer.apple.com/library/content/documentation/General/Conceptual/AppSearch/AppContent.html#//apple_ref/doc/uid/TP40016308-CH7-SW1). With that said, we still have to do [`NSUserActivity` indexing](https://developer.apple.com/library/content/documentation/General/Conceptual/AppSearch/Activities.html#//apple_ref/doc/uid/TP40016308-CH6-SW1) in #8783 so we can make those indexed items public (as appropriate).

**Comment by bummytime** [comment]
@frosty ready for another look. To fix the iPad issues, I ended up just pushing the new `ReaderDetailViewController` onto the detail's navigation stack, thus taking `ReaderMenuViewController` out of the picture. This approach seems to take care of any navigation bar issues after resizing the multitasking panes.

Let me know your thoughts.

**Comment by bummytime** [comment]
Thanks @frosty!

**Pull request merged**